modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/DDMFieldAttributeUpgradeProcess.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.dynamic.data.mapping.internal.upgrade.v5_5_1;
+
+import com.liferay.dynamic.data.mapping.model.DDMStructureVersion;
+import com.liferay.dynamic.data.mapping.service.DDMStructureVersionLocalService;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.dao.orm.common.SQLTransformer;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Alicia García
+ */
+public class DDMFieldAttributeUpgradeProcess extends UpgradeProcess {
+
+	public DDMFieldAttributeUpgradeProcess(
+		DDMStructureVersionLocalService ddmStructureVersionLocalService) {
+
+		_ddmStructureVersionLocalService = ddmStructureVersionLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (PreparedStatement selectPreparedStatement =
+				connection.prepareStatement(
+					SQLTransformer.transform(
+						StringBundler.concat(
+							"select DDMFieldAttribute.ctCollectionId, ",
+							"DDMFieldAttribute.fieldAttributeId, ",
+							"DDMFieldAttribute.fieldId, ",
+							"DDMField.structureVersionId from ",
+							"DDMFieldAttribute inner join DDMField on ",
+							"DDMFieldAttribute.fieldId = DDMField.fieldId ",
+							"where DDMFieldAttribute.companyId = 0")));
+			PreparedStatement updatePreparedStatement =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection,
+					"update DDMFieldAttribute set companyId = ? where " +
+						"ctCollectionId = ? and fieldAttributeId = ?")) {
+
+			ResultSet resultSet = selectPreparedStatement.executeQuery();
+
+			while (resultSet.next()) {
+				long structureVersionId = resultSet.getLong(
+					"structureVersionId");
+
+				DDMStructureVersion ddmStructureVersion =
+					_ddmStructureVersionLocalService.fetchDDMStructureVersion(
+						structureVersionId);
+
+				if (ddmStructureVersion == null) {
+					if (_log.isWarnEnabled()) {
+						_log.warn(
+							"DDMStructureVersion not found for field ID " +
+								resultSet.getLong("fieldId"));
+					}
+				}
+				else {
+					updatePreparedStatement.setLong(
```

Comment:
@AliciaGarciaGarcia, order should follow service.xml 😅. ctCollectionId -> fieldAttributeId -> companyId

Reply:
My bad @AliciaGarciaGarcia